content/optimization/importance.md

```diff
@@ -1,10 +1,284 @@
+&(x_n)_d>\delta \implies\ \text{sample}\ n\ \text{goes to right child node}
+\end{aligned}$$
+
+We emphasize that the decision boundary is drawn by considering a single feature field $d$ and partitioning the $n^\mathrm{th}$ sample by the value at that feature field. Decision boundaries at each internal parent node $t_P$ are formed by choosing a "split criterion," which describes how to partition the set of elements at this node into left and right child nodes $t_L$, $t_R$ with $X_{t_L}\subset X_{t_P}$ and $X_{t_R}\subset X_{t_P}$, $X_{t_L}\cup X_{t_R}=X_{t_P}$. This partitioning is optimal if $X_{t_L}$ and $X_{t_R}$ are pure, each containing only members of the same class. *Impurity measures* are used to evaluate the degree to which the set of data points at a given tree node $t$ are not pure. One common impurity measure is Gini Impurity, 
+
+$$\begin{aligned} 
+I(t) = \sum_{c=1}^C p(c|t)(1-p(c|t))
+\end{aligned}$$
+
+Here, $p(c|t)$ is the probability of drawing a member of class $c$ from the set of elements at node $t$. For example, the Gini impurity at the root node (corresponding to the whole dataset) is 
+
+$$\begin{aligned} 
+I(t_0) = \sum_{c=1}^C \frac{|X_c|}{|X|}(1-\frac{|X_c|}{|X|})
+\end{aligned}$$
+
+In a balanced binary dataset, this would give $I(t_0)=1/2$. If the set at node $t$ is pure, i.e. class labels corresponding to $X_t$ are identical, then $I(t)=0$. We can use $I(t)$ to produce an optimal splitting from parent $t_p$ to children $t_L$ and $t_R$ by defining an *impurity gain*, 
+
+$$\begin{aligned}
+\Delta I = I(t_P) - I(t_L) - I(t_R)
+\end{aligned}$$
+
+This quantity describes the relative impurity between a parent node and its children. If $X_{t_P}$ contains only two classes, an optimal splitting would separate them into $X_{p_L}$ and $X_{p_R}$, producing pure children nodes with $I(t_L)=I(t_R)=0$ and, correspondingly, $\Delta I = I(t_P)$. Accordingly, good splitting decisions should maximize impurity gain. In general, a pure node cannot be split further and must therefore be a leaf. Likewise, a node for which there is no splitting yielding $\Delta I > 0$ must be labeled a leaf. These splitting decisions are made recursively at each node in a tree until some stopping condition is met. Stopping conditions may include maximum tree depths or leaf node counts, or threshhold on the maximum impurity gain. 
+
+Impurity gain gives us insight into the importance of a decision. In particular, larger $\Delta I$ indicates a more important decision. If some feature $(x_n)_d$ is the basis for several decision splits in a decision tree, the sum of impurity gains at these splits gives insight into the importance of this feature. Accordingly, one measure of the feature importance of $d$ as the average (with respect to the total number of internal nodes) impurity gain imparted by decision split on $d$. This method generalizes to the case of BDTs, in which case one would average this quantity across all weak learner trees in the ensemble. 
```

Comment:
"is the average" rather than "as the average"? Also, the parenthetical should go at the end of the sentence.